pyomo/contrib/preprocessing/plugins/int_to_binary.py

```diff
@@ -103,16 +103,18 @@ def _apply_to(self, model, **kwds):
                     "variables." % (int_var.name,)
                 )
             # do the reformulation
-            highest_power = int(floor(log(value(int_var.ub), 2)))
+            highest_power = int(floor(log(value(int_var.ub - int_var.lb), 2)))
             # TODO potentially fragile due to floating point
 
             reform_block.integer_to_binary_constraint.add(
                 idx, expr=int_var == sum(
                     reform_block.new_binary_var[idx, pwr] * (2 ** pwr)
-                    for pwr in range(0, highest_power + 1)))
+                    for pwr in range(0, highest_power + 1))
+                    + int_var.lb)
 
             # Relax the original integer variable
-            int_var.domain = NonNegativeReals
+            # TODO-romeo @qtothec why do we need this?
+            #int_var.domain = NonNegativeReals
```

Comment:
You still have this line commented. You need to have an update from `Integers` to `Reals`, otherwise your model still has integer variables in it. You've just now added binary variables to correspond to each of the integer variables. The purpose of this transformation is to reformulate away all the integers.

Reply:
Oh, I have a misunderstanding of the method then. The way I want to use it is to just introduce the binary variables so that I can do an integer cut. For my use-case, I don't need the variable to be relaxed.

Reply:
I think this is the only blocking issue that I have with this PR, as it breaks functionality for my use cases. If there's no objections, I will simply revert this line and call it a day with this PR.